datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1181,6 +1193,8 @@ pub struct Join {
     pub right: Arc<LogicalPlan>,
     /// Equijoin clause expressed as pairs of (left, right) join columns
     pub on: Vec<(Column, Column)>,
+    /// Filters applied before join output (non-equi conditions)
```

Comment:
```suggestion
    /// Filters applied during join (non-equi conditions)
```

I always thought of these filtering happening "during" the join when matching rows are being found. They aren't really applied on the input (otherwise they could be represented as `Filter`s in the actual plan)